### ISSUE TITLE
Ways to make a file more trustworthy (Signing, submitting to microsoft)

### ISSUE THREAD

**Issue by kaltinril** [vchelaru/Gum#199]
Seems like there are really 3 things.

1. (Chapter 1) unblocking the zip (Fixes the issue) 
2. (Chapter 2-5) signing (Doesn't actually fix the warning, but does show the publisher name instead of unknown)
3. (Chapter 6) Submitting to Microsoft (unsure what it will actually do)

I've put everything into a doc for you
[How to Sign a file.docx](https://github.com/user-attachments/files/17199295/How.to.Sign.a.file.docx)

**Comment by vchelaru** [comment]
Thank you for researching this. I'm going to (currently) postpone working on this because:

1. It seems to be fairly complicated - at least there are many steps
2. There is no guarantee that all of this work will have any impact on the "windows protected your PC" popup. Gum is not super popular so it's possible (likely?) that Microsoft will not accept the certificate and remove that popup.

I'll keep this issue open for now and we can revisit it later.

**Comment by kaltinril** [comment]
Other ways....

Looks like you can also do Windows Installer (If you have a non "express" version of Visual Studio)

Or

You can use the ClickOnce installer (Probably this option)

Click Once
https://learn.microsoft.com/en-us/visualstudio/deployment/clickonce-security-and-deployment?view=vs-2022

Windows Installer
https://learn.microsoft.com/en-us/visualstudio/deployment/installer-projects-net-core?view=vs-2022

**Comment by vchelaru** [comment]
@kaltinril are any of these options something you'd be comfortable trying?

**Comment by vchelaru** [comment]
A recent blog post discussing this. Still seems like a pain, and fairly expensive ($10/month)

https://www.hanselman.com/blog/automatically-signing-a-windows-exe-with-azure-trusted-signing-dotnet-sign-and-github-actions/comments#comments-start

**Comment by vchelaru** [comment]
Thanks again @kaltinril for the thorough research here — splitting this into (1) unblocking the zip, (2) signing, and (3) submitting to Microsoft is exactly the right way to frame it.

After revisiting this, I'm going to close it out. Reasoning, for the record:

**The popup is Windows SmartScreen, and signing alone doesn't reliably remove it.** A standard OV (Organization Validation) certificate is *reputation-based* — trust accrues with download volume rather than being granted the moment you sign — so at Gum's download numbers it could be a long time, or effectively never, before "Windows protected your PC" actually stops appearing. EV (Extended Validation) certs build that reputation faster but run ~$300–600/yr and now require a hardware token, which is overkill for a project this size.

**The only modern path with a sane cost is Azure Trusted Signing** (~$10/month, signs in CI/GitHub Actions). If this is ever revisited, that's the route — but it has its own eligibility requirements (business-entity / identity validation) that would need checking first, and even then there's no hard guarantee it eliminates the prompt at our volume.

**What actually helps newcomers is cheaper than signing.** The real harm is someone seeing a scary popup and bouncing. That's a docs fix, not a signing project: tell users to either click **More info → Run anyway**, or right-click the download → Properties → **Unblock** (which clears the "Mark of the Web" flag Windows stamps on downloaded files). That covers "Chapter 1" of the doc with zero ongoing maintenance. I'll fold it into the setup/download docs.

So: **not pursuing code signing for now** — documenting the unblock step instead — and leaving the door open to Azure Trusted Signing if Gum's reach grows enough to justify the recurring cost and setup. Closing.

**Comment by vchelaru** [comment]
Thanks for the suggestion!

While I see the value here, the implementation cost and complexity are currently a bit high for the project’s immediate roadmap. To keep the "Open Issues" list actionable for active development, I’m going to close this for now.

However, I am still tracking interest! If this is a feature you really need, please leave a comment briefly explaining your specific use case. I monitor new comments on closed threads, and if there is enough active discussion, I will be happy to reopen and re-evaluate this.
